### PR TITLE
Use user saved plan to generate plan details page

### PR DESCRIPTION
**Pull request by timwangmusic**
## Description
Related [issue 359](https://github.com/timwangmusic/Vacation-Planner/issues/359). Note that this is a **breaking** change, since users will not be able to check plan details for the previously saved plans.

## Solution
* Enhance user saved plans to generate the plan details page.
* Set expiration time on system generated plans.

## Verifications
* Save a plan from the result page -> go to details -> save the same plan from there -> verify in the profile that there should be no duplicate
* Save a plan directly from the details page -> go to the result page >  save the same plan from there -> verify in the profile that there should be no duplicate

## Testing
- [x] Integration testing on Heroku testing
- [ ] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: iowrappers/redis_client.go [hunk 1]
```diff
@@ -627,7 +627,7 @@ func (r *RedisClient) SavePlanningSolutions(ctx context.Context, request *Planni
 			Logger.Error(err)
 			continue
 		}
-		_, recordSaveErr := r.client.Set(ctx, solutionRedisKey, json_, 0).Result()
+		_, recordSaveErr := r.client.Set(ctx, solutionRedisKey, json_, PlanningSolutionsExpirationTime).Result()
```
Comment (sananand007) [inline iowrappers/redis_client.go:630]: This was defined on the previous PR I believe

**Review by sananand007 — APPROVED** [review]
lg, did you deploy it, Can I check the test deployment ?

**Comment by timwangmusic** [comment]
> lg, did you deploy it, Can I check the test deployment ? 

I'll deploy to testing, however I think we need to make some tweaks to this change to make the maps working

File: iowrappers/redis_client.go (hunk 1 above)
Comment (timwangmusic) [reply iowrappers/redis_client.go:630]: No, the constant `PlanningSolutionsExpirationTime` existed before if this is what you mean.

**Pull request merged**